jdaviz/configs/specviz/plugins/line_analysis/line_analysis.py

```diff
@@ -332,7 +332,42 @@ def _uncertainty(result):
         for function in FUNCTIONS:
             # TODO: update specutils to allow ALL analysis to take regions and continuum so we
             # don't need these if statements
-            if function == "Equivalent Width":
+            if function == "Line Flux":
+                meta = self.dataset.selected_dc_item.meta
+                telescope = meta.get('TELESCOP',
+                                     meta.get('telescope',
+                                              meta.get('_primary_header', {}).get('TELESCOP', '')))
+                if telescope == 'JWST':
+                    # Perform integration in frequency space
+                    freq_spec = Spectrum1D(spectral_axis=spec_subtracted.spectral_axis.to(u.Hz),
```

Comment:
does this need the spectral equivalency (if the original spectral_axis is in wavelength)?

Reply:
Thanks for the reminder!